File: src/main/java/com/procurement/contracting/repository/ACRepository.java
package com.procurement.contracting.repository;

import com.procurement.contracting.model.entity.ACEntity;
import com.procurement.contracting.model.entity.CANEntity;
import java.util.UUID;
import org.springframework.data.cassandra.repository.CassandraRepository;
import org.springframework.data.cassandra.repository.Query;
import org.springframework.stereotype.Repository;

@Repository
public interface ACRepository extends CassandraRepository<ACEntity, String> {

    @Query(value = "select * from contracting_contract where cp_id=?0 and token_entity=?1 limit 1")
    ACEntity getByCpIdAndToken(String cpId, UUID token);
}

File: src/main/kotlin/com/procurement/contracting/application/repository/ACRepository.kt
package com.procurement.contracting.application.repository

import com.procurement.contracting.domain.entity.ACEntity
import com.procurement.contracting.domain.model.contract.status.ContractStatus
import com.procurement.contracting.domain.model.contract.status.ContractStatusDetails

interface ACRepository {
    fun findBy(cpid: String, contractId: String): ACEntity?

    fun saveCancelledAC(dataCancelledAC: DataCancelledAC)
}

data class DataCancelledAC(
    val id: String,
    val cpid: String,
    val status: ContractStatus,
    val statusDetails: ContractStatusDetails,
    val jsonData: String
)